mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+"""
+This module provides pre-configured HTTP requests
+to consume a desired endpoint.
+"""
+
+from __future__ import annotations
+import sys
+import os
+import logging
+from copy import deepcopy
+from enum import Enum
+from requests import Request, Response, Session
+
+
+class Roles(Enum):
+    """
+    User roles available in McM
+    """
+
+    USER = "user"
+    GENERATOR_CONTACT = "generator_contact"
+    GENERATOR_CONVENER = "generator_convener"
+    PRODUCTION_MANAGER = "production_manager"
+    ADMINISTRATOR = "administrator"
+
+
+class APIRequest:
```

Comment:
This isn't actually a `Request` in the sense that it's not a derived class of `requests.Request`. So the name is  misleading. I think this could be turned into a bunch of `requests.Session` objects and a few private methods in the `McM` class.

Reply:
I agree with this. When I modeled that class, the purpose was to split the responsibilities of consuming endpoints in the databases and the web application and reuse most of the currently available code of the McM REST client. I will merge these tasks in the `McM` class.

Reply:
Already solved, closing this